traitschema/schema.py

```diff
@@ -145,6 +145,14 @@ def to_hdf(self, filename, mode='w', compression=None, compression_opts=None):
                 trait = self.trait(name)
                 # tt = trait.trait_type
 
+                # Workaround for saving arrays containing unicode
+                if trait.array:
+                    data = getattr(self, name)
+                    if str(data.dtype).find("<U") != -1:
```

Comment:
I'd prefer if the two `if` statements are combined, otherwise it's a little awkward to separately use the same `getattr` call in each branch.